argoverse/evaluation/detection_utils.py

```diff
@@ -0,0 +1,175 @@
+        The filtered annotations.
+    """
+    instances = np.array([instance for instance in instances if instance.label_class == target_class])
+
+    if filter_metric == FilterMetric.EUCLIDEAN:
+        centers = np.array([dt.translation for dt in instances])
+        filtered_annos = np.array([])
+
+        if centers.shape[0] > 0:
+            dt_dists = np.linalg.norm(centers, axis=1)
+            filtered_annos = instances[dt_dists < max_dist]
+
+        return filtered_annos
+
+
+def get_ranks(dts: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+    """Get the rankings for the detections.
+
+    Args:
+        dts: Detections.
+
+    Returns:
+        scores: The detection scores.
+        ranks: The ranking for the detections.
+
+    """
+    scores = np.array([dt.score for dt in dts])
+    ranks = scores.argsort()[::-1]
+    return np.expand_dims(scores, 1)[ranks], ranks
+
+
+def interp(prec: np.ndarray, method: InterpType = InterpType.ALL) -> np.ndarray:
+    """Interpolate the precision over all recall levels.
+
+    Args:
+        prec: Precision at all recall levels.
+        method: Accumulation method.
+
+    Returns:
+        Interpolated precision at all recall levels.
+    """
+    if method == InterpType.ALL:
+        prec_interp = np.maximum.accumulate(prec[::-1])[::-1]
+    else:
+        raise NotImplemented("This interpolation method is not implemented!")
+    return prec_interp
+
+
+def compute_match_matrix(dts: np.ndarray, gts: np.ndarray, metric: SimFnType) -> np.ndarray:
+    """Calculate the match matrix between detections and ground truth labels,
+    using a specified similarity function.
+
+    Args:
+        dts: Detections.
+        gts: Ground truth labels.
+        metric: Similarity metric type.
+
+    Returns:
+        Interpolated precision at all recall levels.
+    """
+    if metric == SimFnType.CENTER:
+        dt_centers = np.array([dt.translation for dt in dts])
+        gt_centers = np.array([gt.translation for gt in gts])
+        sims = -cdist(dt_centers, gt_centers)
+    else:
+        raise NotImplemented("This similarity metric is not implemented!")
+    return sims
+
+
+def dist_fn(dts: pd.DataFrame, gts: pd.DataFrame, metric: DistFnType) -> np.ndarray:
+    """Distance functions between detections and ground truth.
+
+    Args:
+        dts: Detections.
+        gts: Ground truth labels.
+        metric: Distance function type.
+
+    Returns:
+        Distance between the detections and ground truth, using the provided metric.
+    """
+    if metric == DistFnType.TRANSLATION:
+        dt_centers = np.vstack(dts["translation"].array)
+        gt_centers = np.vstack(gts["translation"].array)
+        trans_errors = np.linalg.norm(dt_centers - gt_centers, axis=1)
+        return trans_errors
+    elif metric == DistFnType.SCALE:
+        dt_dims = dts[["width", "length", "height"]]
+        gt_dims = gts[["width", "length", "height"]]
+        inter = np.minimum(dt_dims, gt_dims).prod(axis=1)
```

Comment:
i would prefer to make IoU a unit-tested function, e.g.

```python
def iou_aligned_3d():
    inter = np.minimum(dt_dims, gt_dims).prod(axis=1)
    union = np.maximum(dt_dims, gt_dims).prod(axis=1)
    return (inter / union)
```

Reply:
@seanremy 😄